lib/crypto/sig_verifier_test.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2021 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package crypto_test
+
+import (
+	"errors"
+	"io"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/crypto"
+	"github.com/ChainSafe/gossamer/lib/crypto/ed25519"
+	"github.com/ChainSafe/gossamer/lib/crypto/secp256k1"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestVerifySignature(t *testing.T) {
+	t.Parallel()
+
+	errorMessage := errors.New("errors test case")
+	message := []byte("a225e8c75da7da319af6335e7642d473")
+
+	edKeypair, err := ed25519.GenerateKeypair()
+	require.NoError(t, err)
+	edSign, err := edKeypair.Sign(message)
+	require.NoError(t, err)
+
+	secpKeypair, err := secp256k1.GenerateKeypair()
+	require.NoError(t, err)
+	secpSign, err := secpKeypair.Sign(message)
+	require.NoError(t, err)
+
+	srKeypair, err := sr25519.GenerateKeypair()
+	require.NoError(t, err)
+	srSign, err := srKeypair.Sign(message)
+	require.NoError(t, err)
+
+	testCase := map[string]struct {
+		expect             error
```

Comment:
as the finish function just returns `true` or `false` so the `expect` field can be a boolean

Reply:
done!